code/game/atoms.dm

```diff
@@ -1153,3 +1153,10 @@ GLOBAL_LIST_EMPTY(blood_splatter_icons)
 	if(degrees)
 		appearance_flags |= PIXEL_SCALE
 	transform = M
+
+///Setter for the `density` variable to append behavior related to its changing.
+/atom/proc/set_density(new_value)
+	if(density == new_value)
+		return
+	. = density
+	density = new_value
```

Comment:
what *is* this proc?

it returns the previous density only if it changed density, otherwise it returns false

Reply:
If the density is already the same value as you are trying to set it, it does nothing, but if the new value is different to the current value it sets it as that.